app/controllers/homepage_controller.rb

```diff
@@ -2,5 +2,13 @@
 
 # contains all of the methods for the homepage
 class HomepageController < ApplicationController
-  def index; end
+  def index
+    @newbooking = Newbooking.all
+    @totalmoney = 0
+    @totalbookings = 0
+    @newbooking.each do |newbooking|
```

Comment:
remember what I said about `N+1` issues?

In this case, if you have 100 `newbooking` in `@newbooking`, then this would make 100 queries to the DB to get the service for each instance.

You can also make this a little simpler. Ruby enumerables have a method `reduce` which is built for things exactly like this (adding up all of the values). Take a look at: https://medium.com/@james.a.hughes/using-the-reduce-method-in-ruby-907f3c18ae1f

You can also cheat a little on calculating `@totalbookings`. Ruby enumerables also have a `.size` method, which will tell you how many elements there are. That means that you don't need to count them all up - you can just ask `@newbooking.size` to tell you how many there are :)

Reply:
Ok I have fixed this now.